### PR TITLE
chore(docker): set up a local shared postgres for development

### PR DESCRIPTION
**Pull request by orhun**
## Description of change

This PR updates Makefile/`docker-compose` file for creating a local/shared Postgres database called "control DB" for development.

I also realized we already have a shared Postgres at `/dev/tcp/postgres/5432` which begs the question do we still need this control DB?

## How has this been tested? (if applicable)

N/a

### CODE REVIEW

File: docker-compose.dev.yml [hunk 1]
```diff
@@ -30,6 +30,22 @@ services:
   resource-recorder:
     ports:
       - 8007:8000
+    depends_on:
+      - auth
```
Comment (oddgrd) [inline docker-compose.dev.yml:34]: Nit: The .dev.yml file will be combined with the docker-compose.yml into docker-compose.rendered.yml for local development, so we don't need to specify the auth dependency here. I see we did the same thing for logger in this file.

**Review by oddgrd — APPROVED** [review]
Nice! Just one nit.

File: docker-compose.dev.yml [hunk 2]
```diff
@@ -30,6 +30,22 @@ services:
   resource-recorder:
```
Comment (iulianbarbu) [inline docker-compose.dev.yml:30]: I think we can add a similar entry for the `auth` service too, with its dependency on the `control-db`.

**Review by iulianbarbu — APPROVED** [review]

File: docker-compose.dev.yml (hunk 1 above)
Comment (orhun) [reply docker-compose.dev.yml]: 9efbed5845cb18cdfd3071a203abf99a5549ceda

File: docker-compose.dev.yml (hunk 2 above)
Comment (orhun) [reply docker-compose.dev.yml:30]: 5862f38af4b0c5b5fd28b6d8c51adc3025b59134

File: Makefile [hunk 3]
```diff
@@ -73,7 +73,12 @@ DEV_SUFFIX=-dev
 DEPLOYS_API_KEY?=gateway4deployes
 
 # this should use the same version as our prod RDS database
-LOGGER_POSTGRES_TAG?=15
+CONTROL_DB_POSTGRES_TAG?=15
+CONTROL_DB_POSTGRES_PASSWORD?=postgres
+CONTROL_DB_POSTGRES_URI?=postgres://postgres:${CONTROL_DB_POSTGRES_PASSWORD}@control-postgres:5432/postgres
+
+# use the same version as control DB
+LOGGER_POSTGRES_TAG?=${CONTROL_DB_POSTGRES_TAG}
```
Comment (oddgrd) [inline Makefile:81]: While it's highly likely, this shouldn't necessarily be the same version as the control DB.
```suggestion
# this should use the same version as our prod RDS logger database
LOGGER_POSTGRES_TAG?=15
```

**Pull request merged**